praw/models/reddit/rules.py

```diff
@@ -0,0 +1,419 @@
+               print(rule)
+
+        """
+        return iter(self._rule_list)
+
+    @cachedproperty
+    def _rule_list(self) -> List[Rule]:
+        """Get a list of Rule objects.
+
+        :returns: A list of instances of :class:`.Rule`.
+        """
+        rule_list = self._reddit.get(
+            API_PATH["rules"].format(subreddit=self.subreddit)
+        )
+        for rule in rule_list:
+            rule.subreddit = self.subreddit
+        return rule_list
+
+
+class RuleModeration:
+    """Contain methods used to moderate rules.
+
+    To delete ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+    .. code-block:: python
+
+        reddit.subreddit('NAME').rules['No spam'].mod.delete()
+
+    To update ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+    .. code-block:: python
+
+        reddit.subreddit("NAME").removal_reasons["No spam"].mod.update(
+            description="Don't do this!",
+            violation_reason="Spam post")
+    """
+
+    def __init__(self, rule: Rule):
+        """Instantize the RuleModeration class."""
+        self.rule = rule
+
+    def delete(self):
+        """Delete a rule from this subreddit.
+
+        To delete ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+        .. code-block:: python
+
+            reddit.subreddit('NAME').rules['No spam'].mod.delete()
+
+        """
+        data = {
+            "r": str(self.rule.subreddit),
+            "short_name": self.rule.short_name,
+        }
+        self.rule._reddit.post(API_PATH["remove_subreddit_rule"], data=data)
+
+    def update(
+        self,
+        description: Optional[str] = None,
+        kind: Optional[str] = None,
+        short_name: Optional[str] = None,
+        violation_reason: Optional[str] = None,
+    ) -> Rule:
+        """Update the rule from this subreddit.
+
+        .. note:: Existing values will be used for any unspecified arguments.
+
+        :param description: The new description for the rule. Can be empty.
+        :param kind: The kind of item that the rule applies to. One of
+            ``"link"``, ``"comment"``, or ``"all"``.
+        :param short_name: The name of the rule.
+        :param violation_reason: The reason that is shown on the report menu.
+
+        To update ``"No spam"`` from the subreddit ``"NAME"`` try:
+
+        .. code-block:: python
+
+            reddit.subreddit("NAME").removal_reasons["No spam"].mod.update(
+                description="Don't do this!",
+                violation_reason="Spam post")
+
+        """
+        data = {
+            "r": str(self.rule.subreddit),
+            "old_short_name": self.rule.short_name,
+        }
+        for name, value in {
+            "description": description,
+            "kind": kind,
+            "short_name": short_name,
+            "violation_reason": violation_reason,
+        }.items():
+            data[name] = getattr(self.rule, name) if value is None else value
```

Comment:
I may have mentioned this in my previous review, but if not: Does Reddit require that you send all the unchanged values when updating a rule? I assume it does, or else you would not have gone to this effort, but I just wanted to check.